### DISCUSSION TITLE
Is there any difference between ().gfa and ().fastg?

### DISCUSSION THREAD

**Discussion by Kinggerm** [Kinggerm/GetOrganelle#158] [Q&A]
My question: Is there any difference between ().gfa and ().fastq?

_Originally posted by @zgszm in https://github.com/Kinggerm/GetOrganelle/issues/122#issuecomment-1006237056_

**Comment by Kinggerm** [comment]
> My question: Is there any difference between ().gfa and ().fastq?

fastq is the reads file.

I assume your question is on ***.GFA vs *.FASTG**:

 - The *.fastg is the slimmed intermediate assembly graph. It is the data object between `workflow green 4` and `workflow  green 5` @ [main workflow](https://github.com/Kinggerm/GetOrganelle#getorganelle-flowchart)

 - The *.gfa is the assembly result, which is equivalent to *.fasta in sequence content but record connections in some cases (e.g. complete result). It is the data object following the `workflow  green 5`  @ [main workflow](https://github.com/Kinggerm/GetOrganelle#getorganelle-flowchart), i.e. the downstream of *.fastg.

The *.fastg will likely contain more unrelated contigs (False positive hits) because it was generated purely using blast and EM GMM contig clustering and because GetOrganelle was designed to be precautious in removing suspicious contigs in this stage. The generation of *.gfa involves more consideration, including label weighting and path searching, which should be more precise. But the mt-pt similarity/transfers can still largely hamper the precision of mt graph recognition, for which reason some pt contigs were not able to be removed from the mt graph in the *.gfa file. In general, *.gfa file can be organelle-complete and very reliable only when the log file indicates a complete result.

Being a more cautious and more informative graph version that contains potential contaminations and interferences, the *.fastg file will be useful for troubleshooting. 

The *.gfa file is useless in trouble but used for quick confirmation only when the result is already complete.

**References**
https://github.com/Kinggerm/GetOrganelle/wiki/Example-1
Jian-Jun Jin*, Wen-Bin Yu*, Jun-Bo Yang, Yu Song, Claude W. dePamphilis, Ting-Shuang Yi, De-Zhu Li. GetOrganelle: a fast and versatile toolkit for accurate de novo assembly of organelle genomes. Genome Biology 21, 241 (2020). https://doi.org/10.1186/s13059-020-02154-5

**Comment by zgszm** [comment]
Thank you for your answer. My question is solved.
